source/device/routing_module.cpp

```diff
@@ -45,20 +49,24 @@ std::shared_ptr<ILink> RoutingModule::get_link_to_destination(
         return nullptr;
     }
 
-    return (*iterator).second;
+    return (*iterator).second.lock();
 }
 
 std::shared_ptr<ILink> RoutingModule::next_inlink() {
     if (m_inlinks.empty()) {
         LOG_INFO("Inlinks storage is empty");
         return nullptr;
     }
-
-    return *m_next_inlink++;
+    auto inlink = *m_next_inlink++;
+    return inlink.lock();
 }
 
 std::set<std::shared_ptr<ILink>> RoutingModule::get_outlinks() const {
-    return m_outlinks;
+    std::set<std::shared_ptr<ILink>> shared_outlinks;
+    for (auto weak_outlink : m_outlinks) {
+        shared_outlinks.insert(weak_outlink.lock());
+    }
```

Comment:
Lets use `std::transform`:
```suggestion
    correctife_outlinks();
    std::set<std::shared_ptr<ILink>> shared_outlinks;
    std::transform(m_outlinks.begin(), m_outlinks.end(), std::inserter(shared_outlinks), [](auto link) {return link.lock();});
```
Here I use `correctify_outlinks`, but not implement it. It is method similar to `correctify_inlinks` I wrote above